spec/models/resource_score_spec.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe ResourceScore, type: :model do
+  let(:resource) { Resource.first }
+  subject(:resource_score) { FactoryBot.build(:resource_score, resource: resource) }
+
+  describe "validations" do
+    it { is_expected.to be_valid }
+
+    context "uniqueness validation" do
+      before { FactoryBot.create(:resource_score, resource: resource, country: "us", lang: "en") }
+
+      it "validates uniqueness of resource_id scoped to country and lang" do
+        duplicate = FactoryBot.build(:resource_score, resource: resource, country: "us", lang: "en")
+        expect(duplicate).not_to be_valid
+        expect(duplicate.errors[:resource_id]).to include("should have only one score per country and language")
+      end
+    end
+
+    context "featured validation" do
+      it "requires featured_order when featured is true" do
+        resource_score.featured = true
+        resource_score.featured_order = nil
+        expect(resource_score).not_to be_valid
+        expect(resource_score.errors[:featured_order]).to include("must be present if resource is featured")
+      end
+
+      it "validates uniqueness of featured_order within country and language" do
+        ResourceScore.create(resource: resource, featured: true, featured_order: 1, country: "us", lang: "en")
+        duplicate = ResourceScore.new(resource: resource, featured: true, featured_order: 1, country: "us", lang: "en")
+        expect(duplicate).not_to be_valid
+        expect(duplicate.errors[:featured_order]).to include("is already taken for this country and language")
+      end
+
+      it "allows same featured_order for different country" do
+        FactoryBot.create(:resource_score, resource: resource, featured: true, featured_order: 1, country: "US", lang: "en")
+
+        resource2 = Resource.last
+        different_country = FactoryBot.build(:resource_score, resource: resource2, featured: true, featured_order: 1, country: "CA", lang: "en")
+        expect(different_country).to be_valid
+      end
+
+      it "allows same featured_order for different language" do
+        FactoryBot.create(:resource_score, resource: resource, featured: true, featured_order: 1, country: "US", lang: "en")
+
+        resource2 = Resource.last
+        different_lang = FactoryBot.build(:resource_score, resource: resource2, featured: true, featured_order: 1, country: "US", lang: "es")
+        expect(different_lang).to be_valid
+      end
+
+      it "allows same featured_order for different resources" do
+        first_resource = Resource.first
```

Comment:
These can also be moved to let, even if it's not creating the data itself, it's an existing test data piece